src/mem/mem_ctrl.cc

```diff
@@ -1496,7 +1496,8 @@ MemCtrl::MemoryPort::recvFunctional(PacketPtr pkt)
         // calls recvAtomic() and throws away the latency; we can save a
         // little here by just not calculating the latency.
         ctrl.recvFunctional(pkt);
-    }
+    } else 
+        pkt->makeResponse();
```

Comment:
```suggestion
    } else {
        // If the packet is found in the queue, the queue does not makeResponse
        pkt->makeResponse();
    }
```

Reply:
Small change to comment text for clarity.

```suggestion
    } else {
        // The packet's request is satisfied by the queue, but queue does not call makeResponse
        // Here, change the packet to the corresponding response
        pkt->makeResponse();
    }
```